core/include/kernel_defines.h

```diff
@@ -187,6 +187,21 @@ extern "C" {
  */
 #define IS_USED(module) IS_ACTIVE(module)
 
+/**
+ * @def         RIOT_VERSION_NUM(major, minor, patch, extra)
+ * @brief       Generates a 64 bit variable of a release version
+ *              Comparisons to this only apply to released branches
+ *
+ * @param[in]   major   Mayor version of the release
+ * @param[in]   minor   Minor version of the release
+ * @param[in]   patch   Patch level of the release
+ * @param[in]   extra   Extra version, user defined
```

Comment:
How the user can define this and why he should do so should also be documented here. Something like: 


> To define `extra` add a file `EXTRAVERSION` to the RIOT root with the content
> ```makefile
> RIOT_EXTRAVERSION = <extra>
> ```
> with `<extra>` being the number of your local version. This can be useful to ...